### PR TITLE
MAINT: Wrapping some things up

### PR DESCRIPTION
**Pull request by steppi**
This PR wraps up a number of things.

1. It clears up an inconsistency in interpolation methods used for the FFT simpson method between the PDF calculation and the CDF calculation. Previously the variable `pdf_interpolation_kind` was declared but unused in the CDF calculation. My understanding of what was happening is that the PDF calculation used `scipy.interpolate.interp1d` but the CDF calculation used `scipy.interpolate.InterpolatedUnivariateSpline`. The latter was used for the CDF calculation because it makes it easier to integrate over the spline as needed in the CDF calculation. However, `InterpolatedUnivariateSpline` does not take the `kind` parameter, so I think the parameter was just left out. `InterpolatedUnivariateSpline` has an analogous `degree` parameter, which has slightly different behavior from `interp1d`s `kind` parameter. Now both the CDF and PDF use `InterpolatedUnivariateSpline` and `pdf_interpolation_kind` has been renamed to `pdf_interpolation_degree` and takes only integer values like the `degree` parameter of `InterpolatedUnivariateSpline`.
2. The Docstring for `levy_stable_gen` has been revised a little for clarity and some comments have been edited.
3. The change allowing all greek letters to be accepted by the linter has been removed and greek letters have been removed from comments.
4. A test has been added that generates datapoints with `rvs` for different values of `alpha`, `beta`, `gamma`, and `delta` and tests `_fitstart` against these for each of the "S0" and "S1" parametrizations. The goal is to see if fit can recover the original parameters, testing the consistency of  both `fit` and `rvs`. The test fails in all cases. Either `_fitstart` or `rvs` seems to have some remaining issues causing the results to be inaccurate. The test has been marked `xfail`. Perhaps someone should check if there is something wrong with the test. If there's nothing wrong with the test, since the issues with `fit` and `rvs` existed prior to this PR, and because this PR contains many substantial improvements, I think it should still be merged even with this test failing. It can be left for future work to fix things.

### CODE REVIEW

File: scipy/stats/tests/test_distributions.py [hunk 1]
```diff
@@ -3192,6 +3192,31 @@ def test_fit(self):
...
+        data = stats.levy_stable.rvs(
+            alpha, beta, loc=delta, scale=gamma, size=10000, random_state=1234
+        )
+        fit = stats.levy_stable._fitstart(data)
+        alpha_obs, beta_obs, delta_obs, gamma_obs = fit
+        assert_allclose(
+            [alpha, beta, delta, gamma],
+            [alpha_obs, beta_obs, delta_obs, gamma_obs],
+            rtol=0.01,
+        )
```
Comment (ragibson) [inline scipy/stats/tests/test_distributions.py:3218]: Interesting, the alpha and beta estimates here are pretty accurate, but the delta and gamma ones are not. You're right this is maybe related to https://github.com/scipy/scipy/pull/14476 or the usual parameterization confusion of the field.

For a future PR, reading through McCulloch's paper again and doing some extra testing could probably resolve it. In any case, I think this is only used to bootstrap the fit's parameter optimization afterwards, so it's probably not a huge issue regardless.

File: tools/unicode-check.py [hunk 2]
```diff
@@ -10,13 +10,8 @@
 # The set of Unicode code points greater than 127 that we
 # allow in the source code.
 box_drawing_chars = set(chr(cp) for cp in range(0x2500, 0x2580))
-greek_letters = set(chr(cp) for cp in chain(
-    range(0x03b1, 0x3c2), range(0x3c3, 0x03c9 + 1),  # small
-    range(0x0391, 0x03a2), range(0x03a3, 0x03a9 + 1))  # large
-)
 allowed = (set(['®', 'é', 'ê', 'ó', 'ö', 'ü', '∫', '≠', '≥']) |
```
Comment (ragibson) [inline tools/unicode-check.py:13]: Funnily enough, it seems the maintainers do add in these symbols occasionally (I see lambda in the upstream version https://github.com/scipy/scipy/blob/5bfbe973ac849db94eb1bb42f4d51697afeed46e/tools/unicode-check.py#L13-L14), but it's good to see this mass change reverted for the final review.

**Review by ragibson — APPROVED** [review]

File: scipy/stats/tests/test_distributions.py (hunk 1 above)
Comment (steppi) [reply scipy/stats/tests/test_distributions.py:3218]: I see. It probably makes more sense to test fit directly. That can be left for later though.

Comment (lutefiskhotdish) [reply scipy/stats/tests/test_distributions.py:3218]: The functionality of _fitstart() is useful (unofficially) to anyone who has quantiles of a distribution, but not raw samples, and wants to estimate a stable distribution from the quantiles.  That's what I was trying to do that prompted [scipy#14476](https://github.com/scipy/scipy/pull/14476).  But of course that's outside the public API for stable distributions.

**Pull request merged**